packages/yourloops/lib/auth/user.ts

```diff
@@ -0,0 +1,149 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { MedicalData } from "../../models/device-data";
+import { IUser, Preferences, Profile, Settings, Consent, UserRoles } from "../../models/shoreline";
+import config from "../config";
+
+const urlPrefixFromUserRole = {
+  patient: "/patient",
+  hcp: "/professional",
+  caregiver: "/caregiver",
+};
+
+class User implements IUser {
+  userid: string;
+  username: string;
+  role!: UserRoles;
+  emails?: string[];
+  emailVerified?: boolean;
+  profile?: Profile | null;
+  settings?: Settings | null;
+  preferences?: Preferences | null;
+  medicalData?: MedicalData | null;
+  latestConsentChangeDate: Date;
+
+  constructor(id: string, name: string) {
+    this.userid = id;
+    this.username = name;
+    this.latestConsentChangeDate = new Date(0);
+    if (config.LATEST_TERMS !== undefined) {
+      this.latestConsentChangeDate = new Date(config.LATEST_TERMS);
+    }
+  }
+
+  /**
+   * Return the user first name
+   */
+  getFirstName(): string {
+    return this.profile?.firstName ?? "";
+  }
+
+  /**
+   * Return the user last name
+   */
+  getLastName(): string {
+    return this.profile?.lastName ?? this.profile?.fullName ?? this.username;
+  }
+
+  /**
+   * Check If the user should accept is consent at a first login.
+   * @description the first login is determined by null consents object
+   */
+  shouldAcceptConsent(): boolean {
+    if (this?.profile?.termsOfUse === undefined || this?.profile?.termsOfUse === null) {
+      return true;
+    }
+
+    if (this?.profile?.privacyPolicy === undefined || this?.profile?.privacyPolicy === null) {
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Check If the user should renew is consent.
+   */
+  shouldRenewConsent(): boolean {
+    if (this?.profile?.termsOfUse === undefined || this?.profile?.termsOfUse === null) {
+      return true;
+    }
+
+    if (this?.profile?.privacyPolicy === undefined || this?.profile?.privacyPolicy === null) {
+      return true;
+    }
```

Comment:
```suggestion
    if (this.shouldAcceptConsent()) {
      return true;
    }
```
Would have been prettier.

Reply:
yes i know, but for now i did not want to couple the two functions. wanted to wait a little